core/amber/src/main/scala/edu/uci/ics/texera/web/OPversion.java

```diff
@@ -18,48 +17,43 @@ public class OPversion {
             throw new RuntimeException(e);
         }
     }
-    private static Repository repository = git.getRepository();
     private static Properties prop = new Properties();
-    // The folder that is tracked whether is changed or not
-    private static String operatorPath = "core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators";
     // saved path for operator version properties file
     private static String versionPath = currentPath+"/operator_version.properties";
 
-    public static void refreshVersion() throws GitAPIException, IOException{
-        String version;
+    public static String refreshVersion(String operatorName, String operatorPath) throws GitAPIException, IOException{
```

Comment:
Instead of throwing an exception, please wrap it in a try and catch clause

Reply:
done